code/modules/clothing/head/misc_special.dm

```diff
@@ -232,3 +232,79 @@
 /obj/item/clothing/head/cardborg/dropped(mob/living/user)
 	..()
 	user.remove_alt_appearance("standard_borg_disguise")
+
+
+
+/obj/item/clothing/head/collectable/petehat/special
+	name = "super ultra rare Pete's hat!"
+	desc = "It smells faintly of the caribbean."
+	icon_state = "petehat"
+
+	action_button_name = "RHUMBA!!"
+
+	var/uses = 1
+	var/mob/living/carbon/human/pete
+	var/rhumba_duration = 55
+	var/tick = 0
+
+/obj/item/clothing/head/collectable/petehat/special/proc/rhumba(mob/living/carbon/human/M)
+	if(M.head != src)
+		to_chat(M,"You need to be wearing the hat to initiate your Rhumba powers.")
+		return
+	if(uses <= 0)
+		to_chat(M,"You cannot start a Rhumba more than once!")
+		return
+	uses -= 1
+	pete = M
+
+	for(var/O in global_intercoms) //shamelessly copied from the bee briefcase
+		var/obj/item/device/radio/intercom/I = O
+		if(I.z != ZLEVEL_STATION)
+			continue
+		if(!I.on)
+			continue
+		playsound(I, 'sound/effects/cuban_pete.ogg' , 100)
+
+	pete.drop_l_hand()
+	pete.drop_r_hand()
+
+	var/obj/item/toy/cuban_maraca/R = new(pete)
+	var/obj/item/toy/cuban_maraca/L = new(pete)
+
+	pete.put_in_l_hand(L)
+	pete.put_in_r_hand(R)
+
+
+
+	rhumba_proc()
+
+/obj/item/clothing/head/collectable/petehat/special/proc/rhumba_proc()

```

Comment:
some of this seems more appropriate for a process loop than a whitle loop, but eh. Your thoughts @TheDZD